internal/dockerservice/dockerservice.go

```diff
@@ -52,74 +52,93 @@ type Container struct {
 func NewContainer(name string, config container.Config, hostConfig container.HostConfig, networkConfig network.NetworkingConfig) Container {
 	return Container{
 		Name:          name,
-		Ctx:           context.Background(),
 		Config:        config,
 		HostConfig:    hostConfig,
 		NetworkConfig: networkConfig,
 	}
 }
 
-func (d Docker) GetContainer(ctx context.Context, name string) (Container, error) {
-	c := Container{}
-	containers, err := d.Cli.ContainerList(ctx, types.ContainerListOptions{})
+// GetContainer returns a docker container with the provided name (if any exists).
+// if no match exists, it returns a nil container and a nil error.
+func (d Docker) GetContainer(ctx context.Context, name string) (*Container, error) {
+	listFilters := filters.NewArgs()
+	listFilters.Add("name", name)
+	containers, err := d.Cli.ContainerList(ctx, types.ContainerListOptions{Filters: listFilters})
 	if err != nil {
-		return Container{}, fmt.Errorf("error listing containers %w", err)
+		return &Container{}, fmt.Errorf("error listing containers %w", err)
 	}
-	for _, container := range containers {
+	for _, match := range containers {
 		// TODO: name of the container has prefixed with "/"
 		// I have hardcoded here; perhaps there is a better way to handle this
-		if misc.SliceContainsString(container.Names, "/"+name) {
-			c.ID = container.ID
-			c.Config.Image = container.Image
-			break
+		if misc.SliceContainsString(match.Names, "/"+name) {
+			data, err := d.Cli.ContainerInspect(ctx, match.ID)
+			if err != nil {
+				return nil, errors.Wrapf(err, "getting data for container %s", match.ID)
```

Comment:
wondering can't we achieve the same with `%w` verb?

Reply:
I think `%w` is for when the variable is an error (so that it can be wrapped and unwrapped). In this case, we want the ID to be part of the error string so it's okay if we keep it this way.

Reply:
I see. Thanks.